opendatahub-io/opendatahub-operator
# Migration uninstall configmap from DSC to setup controller

## Description


JIRA: [RHOAIENG-16674](https://issues.redhat.com/browse/RHOAIENG-16674)

Created a new setup-controller and moved the below functionalities to uninstallation.go if `HasDeleteConfigMap` exists 
- Deletion of the DataScienceCluster instance
- Operator uninstall 
- Existing workflow 



## How Has This Been Tested?



Testing process : 
- install cluster 
- create dsc and dsci
- create `delete-configmap` with 
```
labels:
    api.openshift.com/addon-managed-odh-delete: 'true'
```
- check dsc and dsci deletion and logs for deletion of other resources

## Screenshot or short clip

<img width="1617" alt="Screenshot 2024-12-16 at 1 00 29 PM" src="https://github.com/user-attachments/assets/a8685285-6683-4a86-960f-8e4738c126c5" />

## Merge criteria
